backends/p4tools/testgen/core/target.h

```diff
@@ -0,0 +1,164 @@
+#ifndef TESTGEN_CORE_TARGET_H_
+#define TESTGEN_CORE_TARGET_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <boost/filesystem.hpp>
+#include <boost/optional/optional.hpp>
+
+#include "backends/p4tools/common/core/solver.h"
+#include "backends/p4tools/common/core/target.h"
+#include "ir/ir.h"
+#include "lib/cstring.h"
+
+#include "backends/p4tools/testgen/core/exploration_strategy/exploration_strategy.h"
+#include "backends/p4tools/testgen/core/program_info.h"
+#include "backends/p4tools/testgen/core/small_step/cmd_stepper.h"
+#include "backends/p4tools/testgen/core/small_step/expr_stepper.h"
+#include "backends/p4tools/testgen/lib/execution_state.h"
+#include "backends/p4tools/testgen/lib/namespace_context.h"
+#include "backends/p4tools/testgen/lib/test_backend.h"
+
+namespace P4Tools {
+
+namespace P4Testgen {
+
+/// Specifies a canonical representation of the target pipeline as documented in P4 code.
+class ArchSpec {
```

Comment:
this could be automated: https://github.com/p4lang/p4c/pull/3281, I have left another comment about this.